sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/SemanticVersion.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.jackson;
+
+import java.util.Objects;
+
+/**
+ * Implements lightweight semantic version based on https://semver.org/ for internal use.
+ */
+final class SemanticVersion implements Comparable<SemanticVersion> {
+    public static  final String UNKNOWN_VERSION = "unknown";
+    public static SemanticVersion getPackageVersionForClass(String className) {
+        Objects.requireNonNull(className, "'className' cannot be null.");
+        try {
+            return SemanticVersion.getPackageVersion(Class.forName(className));
+        } catch (ClassNotFoundException e) {
+            return createInvalid();
+        }
+    }
+
+    public static SemanticVersion parse(String version) {
+        Objects.requireNonNull(version, "'version' cannot be null.");
+        String[] parts = version.split("\\.");
```

Comment:
Here is a full [real semver regex](https://gist.github.com/jhorsman/62eeea161a13b80e39f5249281e17c39) `^([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?` , I'd prefer to keep it simple since we don't really need all parts